### PR TITLE
Rewrite test to use SerializableResource.

### PR DESCRIPTION
**Pull request by beauby**

### CODE REVIEW

File: test/adapter/json_api/resource_type_config_test.rb [hunk 1]
```diff
@@ -27,29 +27,39 @@ def setup
...
-              adapter = ActiveModel::Serializer::Adapter::JsonApi.new(serializer)
-              ActionController::Base.cache_store.clear
-              assert_equal('comment', adapter.serializable_hash[:data][:type])
+            with_adapter :json_api do
+              with_jsonapi_resource_type :singular do
+                hash = ActiveModel::SerializableResource.serialize(@comment).serializable_hash
+                ActionController::Base.cache_store.clear
+                assert_equal('comment', hash[:data][:type])
+              end
             end
```
Comment (bf4) [inline test/adapter/json_api/resource_type_config_test.rb]: or

```ruby
def serialize_with_adapter_and_jsonapi_resource_type(resource, adapter, jsonapi_resource_type)
              with_adapter adapter do
             with_jsonapi_resource_type json_api_resource_type do
                hash = ActiveModel::SerializableResource.serialize(resource).serializable_hash
                ActionController::Base.cache_store.clear
              yield hash
              end
end
```

Comment (beauby) [inline test/adapter/json_api/resource_type_config_test.rb]: I wanted to keep it in line with a possible global `with_adapter` method.

**Comment by beauby** [comment]
Closing in favor of #1093.

**Comment by bf4** [comment]
Any reason you didn't just re-use this PR and force push to test-serializable-resource?  Just curious

**Comment by beauby** [comment]
Hmm good question, I was probably drunk.

**Comment by bf4** [comment]
:beer: ?

**Pull request closed without merging**